k2/csrc/ragged_ops.cu

```diff
@@ -1249,14 +1250,62 @@ std::vector<RaggedShape> GetPrefixes(RaggedShape &src,
 
 Ragged<int32_t> AddSuffixToRagged(Ragged<int32_t> &src,
                                   const Array1<int32_t> &suffix) {
-  K2_LOG(FATAL) << "Not Implemented!";
-  return Ragged<int32_t>();
+  NVTX_RANGE(K2_FUNC);
+  K2_CHECK_EQ(suffix.Dim(), src.TotSize(src.NumAxes() - 2));
+  ContextPtr &c = src.Context();
+  Array1<int32_t> dst_values(c, src.NumElements() + suffix.Dim());
+  int32_t *dst_values_data = dst_values.Data();
+  int32_t num_axes = src.NumAxes();
+  const int32_t *src_row_splits_lastm1_data =
+      src.RowSplits(num_axes - 1).Data(),
+                *src_row_ids_lastm1_data = src.RowIds(num_axes - 1).Data(),
+                *src_values_data = src.values.Data(),
+                *suffix_data = suffix.Data();
+
+  K2_EVAL(
+      c, src.NumElements(), lambda_copy_src, (int32_t idx)->void {
+        int32_t dst_idx = idx + src_row_ids_lastm1_data[idx];
+        dst_values_data[dst_idx] = src_values_data[idx];
+      });
+  K2_EVAL(
+       c, suffix.Dim(), lambda_copy_suffix, (int32_t idx)->void {
+         int32_t row_id = src_row_ids_lastm1_data[idx];
+         int32_t dst_idx = src_row_splits_lastm1_data[row_id + 1] - 1;
```

Comment:
I think 
```cpp
int32_t dst_idx = src_row_splits_lastm1_data[idx + 1] + idx;
```

Reply:
mm, yeah, as @freewym use idx as row id (i.e. idx0) here. Anyway, I suggest we do this in one kenerl indexes with dest_values.Dim() instead of two kenerls

Reply:
Here idx means idx0, so it is incorrect to use it to index `src_row_ids_lastm1_data`.
I would suggest following the naming convention established by Dan.

Reply:
I am not familiar with the naming convention, so I may be wrong: I think  I was using `idx` to index idx012 if `src` is 3D, so `row_id` is idx01. I should have written `dst_idx` as
```c++
int32_t dst_idx = src_row_splits_lastm1_data[row_id + 1] - 1 + row_id;
```

where ` src_row_splits_lastm1_data[row_id + 1] - 1` is the position of the last element of row `row_id` in `src`

Reply:
Please refer to https://github.com/k2-fsa/k2/blob/7e084e9cfd0982c9d2db3db8ff8a2fd6d97e3b6c/k2/csrc/utils.h#L107